libexec/python/shell.py

```diff
@@ -120,9 +120,9 @@ def parse_image_uri(image, uri=None, quiet=False):
     # Now look for registry, namespace, repo
     image = image.split('/')
 
-    if len(image) == 3:
+    if len(image) > 2:
         registry = image[0]
-        namespace = image[1]
+        namespace = "/".join(image[1:-1])
         repo_name = image[2]
 
```

Comment:
Sorry, that's a typo/git foobar on my part, I'm still learning how to build/run in the dev tree and had been working on the installed copy (:scream:).

Just pushed the fix.

FWIW, here's what I'm actually running:

```diff
[vagrant@localhost singularity]$ diff /usr/local/libexec/singularity/python/shell.py libexec/python/shell.py
125c123
<     if len(image) > 2:
---
>     if len(image) == 3:
127,128c125,126
<         namespace = "/".join(image[1:-1])
<         repo_name = image[-1]
---
>         namespace = image[1]
>         repo_name = image[2]
```